packages/common/src/types.ts

```diff
@@ -319,11 +348,25 @@ export interface IHubContent extends IHubResource, IItem {
   data?: {
     [propName: string]: any;
   };
+  // service types: Feature Service, Map Service
+  /** service information (currentVersion, capabilities, maxRecordCount etc) */
+  server?: Partial<IFeatureServiceDefinition>;
+  /** layer information (geometryType, fields, etc) for related layers in the service */
+  layers?: Array<Partial<ILayerDefinition>>;
+  // layer types: Feature Layers, Raster Layers
+  /** layer information (geometryType, fields, etc) */
+  layer?: Partial<ILayerDefinition>;
```

Comment:
Thoughts on just using `layers` and if there is only one entry... treating it like a `layer`?

Reply:
so, `layer` represents the one layer out of all the ones on the server that you are currently looking at.  If this we an instance of a class, we'd have a `layerId` property and a layer getter like `this.layers.filter(layer => layer.id === this.layerId)`.